lib/model/message.dart

```diff
@@ -267,4 +268,17 @@ class MessageStoreImpl with MessageStore {
       view.notifyListenersIfMessagePresent(event.messageId);
     }
   }
+
+  void handleSubmessageEvent(SubmessageEvent event) {
+    final poll = messages[event.messageId]?.poll;
+    if (poll == null) return;
+    poll.applyEvent(
+      event.senderId,
+      PollEvent.fromJson(event.content as Map<String, Object?>)
```

Comment:
It would be challenging, because the expected data structure depends on the type of the widget, and that information comes separately on the Message object.

Reply:
This brings up a possible race. Will it be possible that this event arrives before we have the message?

If it does happen, we wouldn't be able to apply the event. I think the message when it later arrives ought to contain the up-to-date submessages, but I'm not entirely sure.

Reply:
I see. Yeah, the `SubmessageEvent` itself doesn't have a field saying it's for a poll, a todo list, or whatever. Yeah, I guess we'll leave this be.